packages/agent/src/agent/utils/condition-tree-parser.ts

```
@@ -0,0 +1,75 @@
+import {
+  Collection,
+  CollectionUtils,
+  ColumnSchema,
+  ConditionTree,
+  ConditionTreeBranch,
+  ConditionTreeLeaf,
+  Operator,
+} from '@forestadmin/datasource-toolkit';
+
+export default class ConditionTreeParser {
+  static fromPlainObject(collection: Collection, json: unknown): ConditionTree {
+    if (ConditionTreeParser.isLeaf(json)) {
+      const operator = ConditionTreeParser.parseOperator(json.operator);
+      const value = ConditionTreeParser.parseValue(collection, { ...json, operator });
+
+      return new ConditionTreeLeaf(json.field, operator, value);
+    }
+
+    if (ConditionTreeParser.isBranch(json)) {
+      const branch = new ConditionTreeBranch(
+        json.aggregator.toLowerCase() === 'and' ? 'And' : 'Or',
```

if the condition tree from the front passed an unknown aggregator, we convert it to "or". I think we should raise an error.

---

Add tests with "and", "or" and unknown please :)